src/main/java/de/tum/in/www1/artemis/domain/quiz/scoring/ScoringStrategyShortAnswerUtil.java

```diff
@@ -19,7 +17,7 @@ public static int[] getCorrectAndIncorrectSolutionCount(ShortAnswerQuestion shor
         boolean foundCorrectSolution;
         int correctSolutionsCount = 0;
         int incorrectSolutionsCount = 0;
-        List<ShortAnswerSolution> notUsedSolutions = new ArrayList<>(shortAnswerQuestion.getSolutions());
+        HashSet<ShortAnswerSolution> notUsedSolutions = new HashSet<>(shortAnswerQuestion.getSolutions());
```

Comment:
```suggestion
        Set<ShortAnswerSolution> notUsedSolutions = new HashSet<>(shortAnswerQuestion.getSolutions());
```
Always use interface types for variable definitions, not concrete implementations. (Please change that across multiple files in this PR.)